andela/ah-robotics-frontend
# #162949037 Users should be able to Login to Authors Haven

### What does this PR do?

Creates a user Interface through which a user can be able to log in and consumes the User Login API

#### Description of Task to be completed?

- [x] Create a folder structure
- [x] Create React components
- [x] Create React Views
- [x] Design the Login User Interface
- [x] Add UI tests
- [x] Configure Redux and saga middleware
- [x] Consume the Login API
- [x] Add Component tests


### How should this be manually tested?

1. Clone this repository on your local machine using 
```
git clone https://github.com/andela/ah-robotics-frontend.git
````
2. Change your working directory to the project's directory using 
     
```
cd ah-robotics-frontend
```
3. Change to the specific branch `ft-user-login-162949037` using 
```
git checkout ft-user-login-162949037
```

4. Run `npm install` to install all the modules listed in the `package.json` file.

5. Run `npm start` command to start the application.

6. The application should start successfully displaying a page as shown below.
<img width="1280" alt="screen shot 2019-02-14 at 12 24 37" src="https://user-images.githubusercontent.com/14871749/52776855-8e571000-3053-11e9-83e7-f5e1a24da76a.png">

7. On the navigation bar, click login. You will be routed to the login page as shown below:
<img width="1280" alt="screen shot 2019-02-12 at 19 02 56" src="https://user-images.githubusercontent.com/14871749/52649171-ddd4f900-2ef8-11e9-8c58-ac53d81a2077.png">

8. Insert any random email and password in the fields provided.  You will receive relevant error messages as shown in the screenshots section.
To test a successful login, Use the following credentials. You will be redirected to the homepage when you successfully log in.
```
password: o.ogeto@yahoo.com
email: Ogeto1234$
```

9. To run tests, open the terminal and navigate to the project folder where you cloned the repository (as shown in the screenshot below) and run:
```
npm test
```
The following are sample screenshots of the anticipated outputs:
<img width="881" alt="screen shot 2019-02-14 at 12 40 56" src="https://user-images.githubusercontent.com/14871749/52777898-e42cb780-3055-11e9-9246-d0a9b0f90099.png">
<img width="1280" alt="screen shot 2019-02-14 at 12 42 21" src="https://user-images.githubusercontent.com/14871749/52778035-1f2eeb00-3056-11e9-837c-284a3106f4dd.png">


### What are the relevant pivotal tracker stories?

[#162949037](https://www.pivotaltracker.com/n/projects/2235171/stories/162949037)

### Screenshots
#### 1. Error messages on submitting empty data
<img width="1264" alt="screen shot 2019-02-14 at 12 20 14" src="https://user-images.githubusercontent.com/14871749/52777251-674d0e00-3054-11e9-9542-4cb6c183ffa5.png">

Empty password

<img width="1264" alt="screen shot 2019-02-14 at 12 20 49" src="https://user-images.githubusercontent.com/14871749/52777278-7338d000-3054-11e9-8211-0c5b77068da4.png">

#### 2. Error message on invalid credentials
<img width="1263" alt="screen shot 2019-02-14 at 12 21 22" src="https://user-images.githubusercontent.com/14871749/52777322-8c418100-3054-11e9-81f0-2043438648ed.png">
